x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/PhysicalPlanOptimizerTests.java

```diff
@@ -522,12 +553,25 @@ public void testQueryWithAggregation() {
         var exchange = asRemoteExchange(node.child());
         var aggregate = as(exchange.child(), AggregateExec.class);
         assertThat(aggregate.estimatedRowSize(), equalTo(Long.BYTES));
-
         var extract = as(aggregate.child(), FieldExtractExec.class);
         assertThat(names(extract.attributesToExtract()), contains("emp_no"));
         assertThat(aggregate.estimatedRowSize(), equalTo(Long.BYTES));
+
+        var query = source(extract.child());
+        assertThat(query.estimatedRowSize(), equalTo(Integer.BYTES * 2 /* for doc id, emp_no*/));
+        assertThat(query.query(), is(QueryBuilders.existsQuery("emp_no")));
     }
 
+    /**
+     * Expects
+     * EvalExec[[agg_emp{r}#4 + 7[INTEGER] AS x]]
+     * \_LimitExec[500[INTEGER]]
+     *   \_AggregateExec[[],[SUM(emp_no{f}#8) AS agg_emp],FINAL,16]
+     *     \_ExchangeExec[[sum{r}#18, seen{r}#19],true]
+     *       \_AggregateExec[[],[SUM(emp_no{f}#8) AS agg_emp],PARTIAL,8]
+     *         \_FieldExtractExec[emp_no{f}#8]
+     *           \_EsQueryExec[test], query[{"exists":{"field":"emp_no","boost":1.0}}][_doc{f}#34], limit[], sort[] estimatedRowSize[8]
+     */
     public void testQueryWithAggAfterEval() {
```

Comment:
If you could add one more test with two aggregations that show how the `bool` query looks like, that would be great.
For example `from test | stats agg_emp = sum(emp_no), min_s = min(salary)` where the push down should look something like `{"bool":{"should":[{"exists":{"field":"emp_no"}},{"exists":{"field":"salary"}}]}}`